tests/unit/onyxCacheTest.js

```diff
@@ -0,0 +1,680 @@
+import React from 'react';
+import {render} from '@testing-library/react-native';
+
+import waitForPromisesToResolve from '../utils/waitForPromisesToResolve';
+import ViewWithText from '../components/ViewWithText';
+import ViewWithCollections from '../components/ViewWithCollections';
+
+describe('Onyx', () => {
+    describe('Cache Service', () => {
```

Comment:
NAB, appreciate the thoroughness here, but rather than test the individual cache methods we could have just tested the behavior that we'd expect when Onyx is actually using the cache vs. using the cache in isolation.

Reply:
> NAB, appreciate the thoroughness here, but rather than test the individual cache methods we could have just tested the behavior that we'd expect when Onyx is actually using the cache

Actually we've tested both in isolation here and the behavior that we'd expect when Onyx is using cache (in the next section)

It's much harder to test all the expectations written here in Onyx, you have to 
- setup component(s) 
- configure async blocks and flush promises at specific times
- not as straightforward if you want to debug a test

The test for Onyx validate complete flows covering multiple cache methods - e.g. a key is not retrieved from storage when a rendered component uses it